File: bomb/src/main/java/com/cristiancfn/bomb/model/GameSession.java
package com.cristiancfn.bomb.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class GameSession {

    private String roomId;
    private GameStatus status;
    private Long seed;
    private Integer timeRemainingSeconds;
    private Integer currentStrikes;
    private Integer maxStrikes;
    private Integer modulesCount;
    private Integer modulesResolved;
    private String lastActor;

    public enum GameStatus {
        LOBBY,
        IN_PROGRESS,
        DEFUSED,
        EXPLODED
    }
}

File: front-bomb/src/app/models/game-session.ts
export interface GameSession {
    roomId: string;
    status: 'LOBBY' | 'IN_PROGRESS' | 'DEFUSED' | 'EXPLODED';
    seed: number;
    timeRemainingSeconds: number;
    currentStrikes: number;
    maxStrikes: number;
    modulesCount: number;
    modulesResolved: number;
}